packages/devtools/lib/src/memory/memory_inbounds.dart

```diff
@@ -0,0 +1,344 @@
+        InboundsTreeNode('SingleChildrenObjectElement', '_child');
+
+    data = InboundsTreeNode.root()
+      ..addChild(treeNode00
+        ..addChild(treeNode01)
+        ..addChild(treeNode02
+          ..addChild(treeNode03)
+          ..addChild(treeNode04)
+          ..addChild(treeNode05)))
+      ..addChild(treeNode10
+        ..addChild(treeNode11)
+        ..addChild(treeNode12)
+        ..addChild(treeNode13)
+        ..addChild(treeNode14)
+        ..addChild(treeNode15))
+      ..addChild(terryStuff
+        ..addChild(shrineAppState1
+          ..addChild(statefulElement1
+            ..addChild(hashmapEntry1..addChild(singleChildrenObjectElement1))))
+        ..addChild(shrineAppState2
+          ..addChild(statefulElement2
+            ..addChild(
+                hashmapEntry2..addChild(singleChildrenObjectElement2)))));
+  }
+
+  InboundsTreeNode data;
+}
+
+class InboundsTreeNode extends TreeNode<InboundsTreeNode> {
+  InboundsTreeNode(this._name, this._fieldName, [this.instanceHashCode]);
+  InboundsTreeNode.instance(this._instance, [this.instanceHashCode])
+      : _name = _instance.objectRef,
+        _fieldName = '';
+  InboundsTreeNode.root()
+      : _name = 'Instances',
+        _fieldName = '',
+        instanceHashCode = null;
+  InboundsTreeNode.empty()
+      : _name = null,
+        _fieldName = null,
+        instanceHashCode = null;
+
+  String _name;
+
+  InstanceSummary _instance;
+
+  InstanceSummary get instance => _instance;
+
+  String get name => _name;
+
+  /// isNew signals objectRef (object/###) has changed.
+  void setInstance(InstanceSummary theInstance, String hashCode,
+      [bool isNew = false]) {
+    _instance = theInstance;
+    instanceHashCode = hashCode;
+    _name = _name.split(' ')[0]; // Throw away instance objectRef name.
+    _name = (isNew && !isInboundEntry)
+        ? _instance.objectRef
+        : '$name (${instance.objectRef})';
+  }
+
+  final String _fieldName;
+
+  String get fieldName => _fieldName;
+
+  bool get isInboundEntry => _fieldName?.isNotEmpty;
+
+  String instanceHashCode;
+
+  bool get isEmpty =>
+      _name == null && _fieldName == null && instanceHashCode == null;
+}
+
+abstract class InstanceRefsView extends CoreElement {
+  InstanceRefsView(this.inboundsTree) : super('div', classes: 'memory-table');
+
+  final InboundsTreeData inboundsTree;
+
+  bool viewNeedsRebuild = false;
+
+  void rebuildView();
+
+  void reset();
+
+  void update({bool showLoadingSpinner = false}) async {
+    if (inboundsTree == null) return;
+
+    // Update the view if it is visible. Otherwise, mark the view as needing a
+    // rebuild.
+    if (!isHidden) {
+      if (showLoadingSpinner) {
+        final Spinner spinner = Spinner.centered();
+        add(spinner);
+
+        // Awaiting this future ensures the spinner pops up in between switching
+        // profiler views. Without this, the UI is laggy and the spinner never
+        // appears.
```

Comment:
nit: this comment looks like it is from the cpu profiler. Update to reflect the context of the memory page